libs/vgc/ui/menubutton.h

```diff
@@ -0,0 +1,246 @@
+// Copyright 2022 The VGC Developers
+// See the COPYRIGHT file at the top-level directory of this distribution
+// and at https://github.com/vgc/vgc/blob/master/COPYRIGHT
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef VGC_UI_MENUBUTTON_H
+#define VGC_UI_MENUBUTTON_H
+
+#include <string>
+#include <string_view>
+
+#include <vgc/core/innercore.h>
+#include <vgc/geometry/vec2f.h>
+#include <vgc/graphics/richtext.h>
+#include <vgc/ui/action.h>
+#include <vgc/ui/api.h>
+#include <vgc/ui/flex.h>
+#include <vgc/ui/label.h>
+#include <vgc/ui/margins.h>
+#include <vgc/ui/widget.h>
+
+namespace vgc::ui {
+
+VGC_DECLARE_OBJECT(ActionButton);
+VGC_DECLARE_OBJECT(Menu);
+VGC_DECLARE_OBJECT(MenuButton);
+
+/// \class vgc::ui::ActionButton
+/// \brief A clickable widget that represents an action and triggers it on click/hover.
+///
+/// It destroys itself whenever its action gets destroyed.
+///
+class VGC_UI_API ActionButton : public Flex {
+private:
+    VGC_OBJECT(ActionButton, Flex)
+
+protected:
+    ActionButton(Action* action, FlexDirection layoutDirection);
+
+public:
+    /// Creates an MenuButton with the given `action`.
+    ///
+    static ActionButtonPtr
+    create(Action* action, FlexDirection layoutDirection = FlexDirection::Column);
+
+    Action* action() const {
+        return action_;
+    }
+
+    bool isDisabled() const {
```

Comment:
Better change this to isEnabled? That's what Qt does, I think it makes more sense to use the "positive" version.